invenio_users_resources/services/generators.py

```diff
@@ -108,3 +111,31 @@ def query_filter(self, identity=None, **kwargs):
                     return dsl.Q("term", id=need.value)
 
         return []
+
+
+class AdminAction(Generator):
+    """Allows admin-enabled actions."""
+
+    def __init__(self, action):
+        """Constructor."""
+        self.action = action
+        super().__init__()
+
+    def needs(self, **kwargs):
+        """Enabling Needs."""
+        return [self.action]
+
+
+class UserModeration(AdminAction):
+    """Admin action generator for user moderation."""
+
+    def __init__(self):
+        """Constructor."""
+        super().__init__(moderation_action)
+
+    def query_filter(self, identity=None, **kwargs):
```

Comment:
I am not entirely sure about this one ... 
A `UserModeration` should be able to read every user, however, if the generator is used out of the user context then the `MatchAll` query is incorrect.

Reply:
Question: Is the ``query_filter`` needed at all?

Reply:
I think so, otherwise, the moderator can't see other users.

Reply:
Can we use the `excludes` in the default request permission to make an exception when a user can moderate?

Reply:
What is the `default request permission` in this case? 

The idea is that a user moderator can see every user in the system (be it public, private, active or not, deleted or not.) So, in the users' search, the user moderator has the same permission as a "system process" (e.g. they can see everything). 

That's also why I split the base generator (`AllowedByAction`) from the user moderator, just to make sure that the user moderator has the right search filters ONLY in the users' context.